A part of:
  http://sourceware.org/ml/gdb-cvs/2008-06/msg00160.html

It was crashing the RH testcase `gdb.threads/watchthreads2.exp'.

diff -up -u -X /root/jkratoch/.diffi.list -rup gdb-6.8/gdb/breakpoint.c gdb-6.8-patched/gdb/breakpoint.c
--- gdb-6.8/gdb/breakpoint.c	2008-08-28 11:56:56.000000000 +0200
+++ gdb-6.8-patched/gdb/breakpoint.c	2008-08-28 11:53:43.000000000 +0200
@@ -3173,6 +3173,9 @@ bpstat_what (bpstat bs)
 	/* I suspect this can happen if it was a momentary breakpoint
 	   which has since been deleted.  */
 	continue;
+      if (bs->breakpoint_at->owner == NULL)
+	bs_class = bp_nostop;
+      else
       switch (bs->breakpoint_at->owner->type)
 	{
 	case bp_none:
